packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -496,6 +503,335 @@ describe('TransactionController Integration', () => {
+                params: ['0x6bf137f335ea1b8f193b8f6ea92561a60d23a207', '0x1'],
+              },
+              response: {
+                result: '0x1',
+              },
+            },
+            {
+              request: {
+                method: 'eth_sendRawTransaction',
+                params: [
+                  '0x02e383aa36a7010101019408f137f335ea1b8f193b8f6ea92561a60d23a2118080c0808080',
+                ],
+              },
+              response: {
+                result: '0x1',
+              },
+            },
+            {
+              request: {
+                method: 'eth_blockNumber',
+                params: [],
+              },
+              response: {
+                result: '0x3',
+              },
+            },
+            {
+              request: {
+                method: 'eth_blockNumber',
+                params: [],
+              },
+              response: {
+                result: '0x3',
+              },
+            },
+            {
+              request: {
+                method: 'eth_getTransactionReceipt',
+                params: ['0x1'],
+              },
+              response: {
+                result: {
+                  blockHash: '0x1',
+                  blockNumber: '0x3', // we need at least 2 blocks mocked since the first one is used for when the blockTracker is instantied before we have listeners
+                  status: '0x1', // 0x1 = success
+                },
+              },
+            },
+          ],
+        });
+        const { transactionController, approvalController } =
+          await newController({});
+        const firstTransaction =
+          await transactionController.addTransaction(
+            {
+              from: ACCOUNT_MOCK,
+              to: ACCOUNT_2_MOCK,
+            },
+            { networkClientId: 'goerli' },
+          );
+        const secondTransaction = await transactionController.addTransaction(
+          {
+            from: ACCOUNT_MOCK,
+            to: ACCOUNT_2_MOCK,
+          },
+          { networkClientId: 'sepolia', origin: 'test' },
+        );
+
+        await Promise.all([
+          approvalController.accept(firstTransaction.transactionMeta.id),
+          approvalController.accept(secondTransaction.transactionMeta.id),
+        ]);
+        await advanceTime({ clock, duration: 1 });
+        await advanceTime({ clock, duration: 1 });
+
+        await Promise.all([firstTransaction.result, secondTransaction.result]);
+
+        // blocktracker polling is 20s
+        await advanceTime({ clock, duration: 20000 });
+        await advanceTime({ clock, duration: 1 });
+        await advanceTime({ clock, duration: 1 });
+
+        expect(transactionController.state.transactions).toHaveLength(2);
+        expect(transactionController.state.transactions[0].status).toBe(
+          'confirmed',
+        );
+        expect(transactionController.state.transactions[1].status).toBe(
+          'confirmed',
+        );
```

Comment:
[nit] could we assert that these were executed on the correct chainIds/networkClients?